viabel/approximations.py

```diff
@@ -372,3 +374,108 @@ def supports_pth_moment(self, p):
     def df(self):
         """Degrees of freedom."""
         return self._df
+
+
+class NeuralNet(ApproximationFamily):
+    def __init__(self, layers_shapes, nonlinearity = np.tanh):
+        self._pattern = PatternDict(free_default = True)
+        self._layers = len(layers_shapes)
+        self._nonlinearity = nonlinearity
+        self.input_dim = layers_shapes[1][1]
+        for l in range(len(layers_shapes)):
+            self._pattern[str(l)] = NumericArrayPattern(shape = layers_shapes[l])
+
+    def forward(self, var_param, x):
+        for l in range(self._layers):
+            W = var_param[str(l)]
+            x = np.dot(x, W)
+            x = np.where(x < 0, 1e-2 * x, x)
+        return self._nonlinearity(x)
+
+    def sample(self, var_param, n_samples):
+        x = npr.multivariate_normal(mean = [0] * self.input_dim,
+                                    cov = np.identity(self.input_dim),
+                                    size = n_samples)
+        z = self.forward(x)
+        return z
+
+    def log_density(self, var_param, x):
+        pass
+
+    def mean_and_cov(self, var_param):
+        samples = self.sample(var_param, 200000)
```

Comment:
make number of MC samples an option in the constructor and set default to something smaller (maybe 10K?)